### PR TITLE
fix: change input values

### PR DESCRIPTION
**Pull request by KimJaea**
I changed input values from arrays of vector to arrays of arrays of vectors.

### CODE REVIEW

File: GOOD-I-DEER/nodes/cosine-similarity-node/calculate-cosine.js [hunk 1]
```diff
@@ -2,30 +2,43 @@ module.exports = function (RED) {
...
+        for (let i = 0; i < vectorA.length; i++) {
+          dotProduct += vectorA[i] * vectorB[i];
+          magnitudeA += vectorA[i] * vectorA[i];
+          magnitudeB += vectorB[i] * vectorB[i];
+        }
+        magnitudeA = Math.sqrt(magnitudeA);
+        magnitudeB = Math.sqrt(magnitudeB);
+
+        if (magnitudeA === 0 || magnitudeB === 0) {
+          throw new Error("Can't calculate cosine similarity of 0 Vector.");
```
Comment (Gitgloo) [inline GOOD-I-DEER/nodes/cosine-similarity-node/calculate-cosine.js]: If an error occurs, node-red terminates. 😥

**Review by chickennight — DISMISSED** [review]
well done change, LGTM 👍

File: GOOD-I-DEER/nodes/cosine-similarity-node/calculate-cosine.js [hunk 2]
```diff
@@ -40,25 +53,23 @@ module.exports = function (RED) {
...
-      var input_vector = msg.payload;
-      var stored_vector = await getStoredVector();
+      var input_vectors = JSON.parse(msg.payload);
+      var stored_vectors = await getStoredVector();
 
-      console.log(input_vector.length);
-      console.log(Object.keys(stored_vector).length);
       var result = 1;
-      if (typeof input_vector == "undefined" || input_vector == null || input_vector == "") {
+      if (typeof input_vectors == "undefined" || input_vectors == null || input_vectors == "") {
```
Comment (hy06ix) [inline GOOD-I-DEER/nodes/cosine-similarity-node/calculate-cosine.js]: Can't we replace this part with "if (!input_vectors) "?

File: GOOD-I-DEER/nodes/cosine-similarity-node/calculate-cosine.js [hunk 3]
```diff
@@ -40,25 +53,23 @@ module.exports = function (RED) {
...
         console.error("Input vector is not valid.");
       } else if (
-        typeof stored_vector == "undefined" ||
-        stored_vector == null ||
-        stored_vector == ""
+        typeof stored_vectors == "undefined" ||
```
Comment (hy06ix) [inline GOOD-I-DEER/nodes/cosine-similarity-node/calculate-cosine.js]: Same question like line 60

**Review by hy06ix — DISMISSED** [review]
Good change from single vector input to multiple vector input 👍

**Review by hy06ix — DISMISSED** [review]

File: GOOD-I-DEER/nodes/cosine-similarity-node/calculate-cosine.js (hunk 1 above)
Comment (leecr1215) [reply GOOD-I-DEER/nodes/cosine-similarity-node/calculate-cosine.js]: because of throwing err

Comment (KimJaea) [reply GOOD-I-DEER/nodes/cosine-similarity-node/calculate-cosine.js]: Thanks, I fixed it. 👍 Please check recent commit below.

File: GOOD-I-DEER/nodes/cosine-similarity-node/calculate-cosine.js [hunk 4]
```diff
@@ -70,10 +72,16 @@ module.exports = function (RED) {
...
         }
-  
-        msg.payload = result;
-        msg.topid = "Result";
-        this.send(msg);  
+
+        if (!result) {
+          msg.payload = "The Vector is not available to calculate.";
+          msg.topic = "Error";
+          this.send(msg);
```
Comment (ramaking) [inline GOOD-I-DEER/nodes/cosine-similarity-node/calculate-cosine.js]: This code may cause unexpected type errors in the node responsible for output.
Don't use send method, just use node.error method and finish the node.

File: GOOD-I-DEER/nodes/cosine-similarity-node/calculate-cosine.js [hunk 5]
```diff
@@ -70,10 +72,16 @@ module.exports = function (RED) {
...
           result = devideSimilarity(input_vectors, stored_vectors);
         }
-  
-        msg.payload = result;
-        msg.topid = "Result";
-        this.send(msg);  
+
+        if (!result) {
+          msg.payload = "The Vector is not available to calculate.";
+          msg.topic = "Error";
```
Comment (Gitgloo) [inline GOOD-I-DEER/nodes/cosine-similarity-node/calculate-cosine.js]: why msg has 'topic' value?

File: GOOD-I-DEER/nodes/cosine-similarity-node/calculate-cosine.js [hunk 6]
```diff
@@ -70,10 +72,16 @@ module.exports = function (RED) {
...
+          this.send(msg);
+        } else {
+          msg.payload = result;
+          msg.topid = "Result";
```
Comment (Gitgloo) [inline GOOD-I-DEER/nodes/cosine-similarity-node/calculate-cosine.js]: I think 'topic' is right, not 'topid'

File: GOOD-I-DEER/nodes/cosine-similarity-node/calculate-cosine.js (hunk 4 above)
Comment (chickennight) [reply GOOD-I-DEER/nodes/cosine-similarity-node/calculate-cosine.js]: it will be better use 'node.error' in error case

Comment (KimJaea) [reply GOOD-I-DEER/nodes/cosine-similarity-node/calculate-cosine.js]: Checked! Now this handle every error by "this.error()". Please check new commit below 😄

File: GOOD-I-DEER/nodes/cosine-similarity-node/calculate-cosine.js (hunk 5 above)
Comment (KimJaea) [reply GOOD-I-DEER/nodes/cosine-similarity-node/calculate-cosine.js]: I used this to distinguish correct return value and others. But now I deleted it.

**Review by Gitgloo — APPROVED** [review]

**Review by chickennight — APPROVED** [review]

**Pull request merged**